esteid/mobileid/types.py

```diff
@@ -38,3 +47,27 @@
         "certificate",
     ],
 )
+
+
+PHONE_NUMBER_REGEXP = re.compile(r"^\+37[02]\d{7,8}$")  # Mobile ID supports Estonian and Lithuanian phones
+
+
+class UserInput(PredictableDict):
+    phone_number: str
+    id_code: str
+    language: Optional[str]
+
+    def is_valid(self, raise_exception=True):
+        result = super().is_valid(raise_exception=raise_exception)
+        if result:
+            if not (self.phone_number and re.match(PHONE_NUMBER_REGEXP, self.phone_number)):
+                raise InvalidParameter(param="phone_number")
```

Comment:
I don't think this should be validated by esteid as this regex based approach might be too fragile. I would remove the phone number validation from this method and would just expect users to pass in validated values.

> I hope the MID/SmartID rest api will raise an error in case the provided data is not valid.

Reply:
I don't think the phone numbering scheme for Estonia and Lithuania is going to change soon :)
I could make the regexp accept broader values (like 7-10 digits maybe), but I'd leave at least some sanity check in place. Otherwise this could result in us being a mediator in DDoS attacks :)

Reply:
I would still leave that to be something to deal with in the userland. Phone number validation is not really the scope of this project.

Reply:
The problem is that userland can also make mistakes and forget about sanity checks. If we don't do a sanity check, we're still open to trouble. Also as we do ID code validation, it would be logical to do at least _some_ phone validation. Phones will always consist of `+`, country code , and several digits, the question is how many digits :) If MID starts supporting other countries, I guess it could be adjusted.
This is by no means a complete validation, I'd prefer to call it sanity check.

Reply:
What trouble are you worried about exactly? This is just an api call which has validation on their side anyway.

In the end I would also be happy if it would be possible to easily override this validation in the userland (currently I can't even monkeypatch this as all validation occurs in the same method).

Reply:
I'm worried that potential requests with invalid data can lead to a) performance degradation of the app itself because MID is not the fastest responding service b) problems with MID as provider that we (i.e. the client) are not performing the necessary validation.

I was just about to suggest that I could make it optional, adding a feature switch. This would help obviate potential issues with it.

Reply:
Or just make this `PHONE_NUMBER_REGEXP` configurable, what do you think?

Reply:
Created a PR #53 with this check made optional and configurable.